pkg/sql/hints/hint_cache.go

```diff
@@ -357,6 +376,30 @@ var _ rangefeedbuffer.Event = &bufferEvent{}
 // Reading from the cache.
 // ============================================================================
 
+// Await blocks until the StatementHintsCache's rangefeed watcher catches up
+// with the present. After Await returns, MaybeGetStatementHints should
+// accurately reflect all hints that were modified before the call to Await
+// (assuming the ctx was not canceled).
+func (c *StatementHintsCache) Await(ctx context.Context, st *cluster.Settings) {
+	// The frontier timestamp comes from the rangefeed, and could be up to
+	// kv.closed_timestamp.target_duration +
+	// kv.rangefeed.closed_timestamp_refresh_interval behind the present.
+	targetDuration := closedts.TargetDuration.Get(&st.SV)
+	refreshInterval := kvserver.RangeFeedRefreshInterval.Get(&st.SV)
+	const fudge = 10 * time.Millisecond
+	waitUntil := c.clock.Now().AddDuration(targetDuration + refreshInterval + fudge).WallTime
+
+	// Await is only used for testing, so we don't need to wake up immediately. We
+	// can get away with polling the frontier time.
+	for frontier := c.frontier.Load(); frontier < waitUntil; frontier = c.frontier.Load() {
+		select {
+		case <-ctx.Done():
+			return
+		case <-time.After(time.Duration(waitUntil-frontier) * time.Nanosecond):
```

Comment:
If the rangefeed is lagging by a larger duration than expected, could we end up looping with no delay?